internal/kibana/saved_objects.go

```diff
@@ -107,3 +109,146 @@ func (c *Client) findDashboardsNextPage(page int) (*savedObjectsResponse, error)
 	}
 	return &r, nil
 }
+
+// SetManagedSavedObject method sets the managed property in a saved object.
+// For example managed dashboards cannot be edited, and setting managed to false will
+// allow to edit them.
+// Managed property cannot be directly changed, so we modify it by exporting the
+// saved object and importing it again, overwriting the original one.
+func (c *Client) SetManagedSavedObject(savedObjectType string, id string, managed bool) error {
+	exportRequest := ExportSavedObjectsRequest{
+		ExcludeExportDetails:  true,
+		IncludeReferencesDeep: false,
+		Objects: []ExportSavedObjectsRequestObject{
+			{
+				ID:   id,
+				Type: savedObjectType,
+			},
+		},
+	}
+	objects, err := c.ExportSavedObjects(exportRequest)
+	if err != nil {
+		return fmt.Errorf("failed to export %s %s: %w", savedObjectType, id, err)
+	}
+
+	for _, o := range objects {
+		o["managed"] = managed
+	}
+
+	importRequest := ImportSavedObjectsRequest{
+		Overwrite: true,
+		Objects:   objects,
+	}
+	_, err = c.ImportSavedObjects(importRequest)
+	if err != nil {
+		return fmt.Errorf("failed to import %s %s: %w", savedObjectType, id, err)
+	}
+
+	return nil
+}
+
+type ExportSavedObjectsRequest struct {
+	ExcludeExportDetails  bool                              `json:"excludeExportDetails"`
+	IncludeReferencesDeep bool                              `json:"includeReferencesDeep"`
+	Objects               []ExportSavedObjectsRequestObject `json:"objects"`
+}
+
+type ExportSavedObjectsRequestObject struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
+
+func (c *Client) ExportSavedObjects(request ExportSavedObjectsRequest) ([]map[string]any, error) {
+	body, err := json.Marshal(request)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode request: %w", err)
+	}
+
+	path := SavedObjectsAPI + "/_export"
```

Comment:
Good point. In https://github.com/elastic/elastic-package/pull/1357 we used the new API only with newer versions of Kibana to avoid possible regressions.

But I have checked and this API is available in all the versions of Kibana we support. And this is a new feature, so there is no risk to introduce regressions, and better to use the most recent API, even if it is in preview.

Note: The used API is in preview, but the other possible APIs are deprecated, so better to use the new one so the old ones can be removed at some point.